### PR TITLE
MoveAllFilesRecursively() works the same on all platforms

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/177628756

These fix ensures that we can copy artifact files in the target installation directory, even if they already exist (with read-only permissions on Windows).

### CODE REVIEW

**Review by MDrakos — APPROVED** [review]

File: internal/fileutils/fileutils.go [hunk 1]
```diff
@@ -498,6 +498,12 @@ func MoveAllFilesRecursively(fromPath, toPath string, cb MoveAllFilesCallback) e
 				return errs.Wrap(err, "os.Remove %s failed", subFromPath)
 			}
 		} else {
+			// If the subToPath file exists, we remove it first - to ensure compatibility between in platforms:
+			// On Windows, the renaming step can otherwise fail if subToPath is read-only (file removal is allowed)
+			if toPathExists {
+				err = os.Remove(subToPath)
+				logging.Error("Failed to remove destination file %s: %v", subToPath, err)
+			}
```
Comment (Naatan) [inline internal/fileutils/fileutils.go:506]: Can we take this chance to clean up the conditional flow here a little? I'm confused why this isn't in the conditional on line 482, I'm guessing it's got something to do with it being a file, but the flow of this logic is now making it really hard to grok why the code lives where it lives.

Not asking for any major refactoring, just clean up the conditionals so it's more self-explanatory.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/fileutils/fileutils.go (hunk 1 above)
Comment (mdrohmann) [reply internal/fileutils/fileutils.go:506]: Yes, the block starting on line 482 is still handling with files AND directories, just checking whether the arguments are compatible.  But, yes I think I can flatten this logic a little...

**Review by Naatan — APPROVED** [review]
Much better! This function is still RIPE for a cleanup, but let's not get too distracted for this story :p

**Comment by mdrohmann** [comment]
@Naatan Could you please merge this to master?  It is still failing on the TestCarlisle integration test, but that should work again once Mike's PR is also merged...

**Pull request merged**